File: apps/macos/Sources/ConfigManager.swift
import Foundation
import VibeProxyCore

/// Manages application configuration using shared core
class ConfigManager {
    static let shared = ConfigManager()
    
    private var appConfig: AppConfig?
    private let configURL: URL
    
    private init() {
        let appSupport = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first!
        let configDir = appSupport.appendingPathComponent("VibeProxy", isDirectory: true)
        
        // Create directory if it doesn't exist
        try? FileManager.default.createDirectory(at: configDir, withIntermediateDirectories: true)
        
        configURL = configDir.appendingPathComponent("config.json")
    }
    
    /// Load configuration from shared core default or file
    func loadConfig() -> AppConfig {
        // Try to load from file first
        if let data = try? Data(contentsOf: configURL),
           let config = try? JSONDecoder().decode(AppConfig.self, from: data) {
            appConfig = config
            return config
        }
        
        // Fall back to shared core default
        let defaultConfig = VibeProxyCore.defaultConfig()
        appConfig = defaultConfig
        return defaultConfig
    }
    
    /// Save configuration to file
    func saveConfig(_ config: AppConfig) throws {
        let encoder = JSONEncoder()
        encoder.outputFormatting = [.prettyPrinted, .sortedKeys]
        let data = try encoder.encode(config)
        try data.write(to: configURL)
        appConfig = config
    }
    
    /// Get current configuration
    func getConfig() -> AppConfig {
        if let config = appConfig {
            return config
        }
        return loadConfig()
    }
    
    /// Update backend configuration
    func updateBackend(_ backend: BackendConfig) throws {
        var config = getConfig()
        config.backend = backend
        try saveConfig(config)
    }
    
    /// Update SLM configuration
    func updateSLM(_ slm: SLMConfig) throws {
        var config = getConfig()
        config.slm = slm
        try saveConfig(config)
    }
    
    /// Update tunnel configuration
    func updateTunnel(_ tunnel: TunnelConfig) throws {
        var config = getConfig()
        config.tunnel = tunnel
        try saveConfig(config)
    }
}

File: apps/windows/VibeProxy/ConfigManager.cs
using System;
using System.IO;
using System.Text.Json;
using VibeProxyCore;

namespace VibeProxy;

/// <summary>
/// Manages application configuration using shared core
/// </summary>
public class ConfigManager
{
    private static ConfigManager? _instance;
    public static ConfigManager Instance => _instance ??= new ConfigManager();
    
    private AppConfig? _appConfig;
    private readonly string _configPath;
    
    private ConfigManager()
    {
        var appData = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
        var configDir = Path.Combine(appData, "VibeProxy");
        Directory.CreateDirectory(configDir);
        _configPath = Path.Combine(configDir, "config.json");
    }
    
    /// <summary>
    /// Load configuration from file or shared core default
    /// </summary>
    public AppConfig LoadConfig()
    {
        // Try to load from file first
        if (File.Exists(_configPath))
        {
            try
            {
                var json = File.ReadAllText(_configPath);
                _appConfig = JsonSerializer.Deserialize<AppConfig>(json);
                if (_appConfig != null)
                {
                    return _appConfig;
                }
            }
            catch
            {
                // Fall through to default
            }
        }
        
        // Fall back to shared core default
        try
        {
            _appConfig = VibeProxyCore.VibeProxyCore.GetDefaultConfig();
            return _appConfig;
        }
        catch
        {
            // Last resort: create default manually
            _appConfig = new AppConfig();
            return _appConfig;
        }
    }
    
    /// <summary>
    /// Save configuration to file
    /// </summary>
    public void SaveConfig(AppConfig config)
    {
        var options = new JsonSerializerOptions { WriteIndented = true };
        var json = JsonSerializer.Serialize(config, options);
        File.WriteAllText(_configPath, json);
        _appConfig = config;
    }
    
    /// <summary>
    /// Get current configuration
    /// </summary>
    public AppConfig GetConfig()
    {
        return _appConfig ?? LoadConfig();
    }
}
